.github/workflows/pull_request_fc.yml

```diff
@@ -30,42 +35,48 @@ jobs:
+            patch -p0 < aarch64_config.patch
+            mv microvm-kernel-arm64.config arch/arm64/configs/microvm.config
+          else
+            wget https://raw.githubusercontent.com/firecracker-microvm/firecracker/master/resources/microvm-kernel-x86_64.config -O arch/x86/configs/microvm.config
+          fi
           touch .config
           make microvm.config
           make vmlinux -j$(nproc)
+          echo ::set-output name=build_exit_code::$?
 
       - name: Build virtio-accel
+        id: build-virtio
         working-directory: virtio-accel
-        run: make KDIR=${GITHUB_WORKSPACE}/linux ZC=0
+        run: |
+          make KDIR=../linux ZC=0
+          echo ::set-output name=build_exit_code::$?
 
-      - name: Upload artifact
-        uses: actions/upload-artifact@v2
+      - name: Upload vmlinux
+        id: upload-vmlinux
+        if: ${{ (steps.build-vmlinux.outputs.build_exit_code == 0 && steps.build-virtio.outputs.build_exit_code == 0) }}
+        uses: cloudkernels/minio-upload@master
         with:
-          name: virtio-accel-${{ github.event.pull_request.head.sha }}
-          path: |
-            linux/vmlinux
-            virtio-accel/virtio_accel.ko
-
-  upload_artifact:
-    needs: build
-    runs-on: ubuntu-latest
-    container:
-      image: minio/mc
-      env:
-        NBFC_S3_ACCESS: ${{ secrets.AWS_ACCESS_KEY }}
-        NBFC_S3_SECRET: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          url: https://s3.nubificus.co.uk
+          access-key: ${{ secrets.AWS_ACCESS_KEY }}
+          secret-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          local-path: linux/vmlinux
+          remote-path: nbfc-assets/github/virtio-accel/${{ matrix.arch }}/${{ github.event.pull_request.head.sha }}/linux/
 
-    steps:
-      - name: Download artifact
-        uses: actions/download-artifact@v2
+      - name: Upload virtio
+        id: upload-virtio
+        if: ${{ (steps.build-vmlinux.outputs.build_exit_code == 0 && steps.build-virtio.outputs.build_exit_code == 0) }}
+        uses: cloudkernels/minio-upload@master
         with:
-          name: virtio-accel-${{ github.event.pull_request.head.sha }}
-
-      - name: Upload artifact to s3
-        run: |
-          mc alias set nbfc-rook https://s3.nubificus.co.uk $NBFC_S3_ACCESS $NBFC_S3_SECRET
-          mc cp -r linux nbfc-rook/nbfc-assets/github/virtio-accel/${{ github.event.pull_request.head.sha }}/
-          mc cp -r virtio-accel/virtio_accel.ko nbfc-rook/nbfc-assets/github/virtio-accel/${{ github.event.pull_request.head.sha }}/linux/
+          url: https://s3.nubificus.co.uk
+          access-key: ${{ secrets.AWS_ACCESS_KEY }}
+          secret-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          local-path: virtio-accel/virtio_accel.ko
+          remote-path: nbfc-assets/github/virtio-accel/${{ matrix.arch }}/${{ github.event.pull_request.head.sha }}/linux/
```

Comment:
Could you make it, so the SHA appears in the path before the architecture, so it is the same as with the other sub-modules?